### ISSUE TITLE
Suppress unused warnings for auto property with serialised backing field

### ISSUE THREAD

**Issue by citizenmatt** [JetBrains/resharper-unity#1949]
Given an auto property such as:

```
[field:SerializeField]
public string UsageHotkey { get; private set; }
```

Rider/ReSharper will mark the setter as unused, because Unity will initialise the backing field directly. Similarly, given a getter only property, Rider/ReSharper will warn that the property is never assigned.

Arguably, the getter only version is more appropriate - a private setter is genuinely unused in the code, so it's reasonable to allow it to be highlighted. We should suppress the warning on the getter only property, although it will be important to check that this is a ReSharper warning and not a compiler warning (we can't and shouldn't suppress a compiler warning).

**Comment by JFletcher94** [comment]
This can potentially cause a user to make a mistake that results in lost work. If a user does the following:
- Create a `[field:SerializeField]` auto property as described above
- Provide a value for the property in the Unity Editor
- Remove `private set;` from the auto property based on the `Auto-property accessor is never used` tooltip in Rider/ReSharper

The Unity Editor loses access to the property, so **the serialized field disappears, along with whatever value it was set to**. If the user then adds the `private set;` back to the auto property, the field in the Unity Editor will be restored but the **previously assigned value will not**.

The values stored in serialized fields may be valuable and hard to recover once lost. Given the potential lost work, and the non-obvious nature of this behavior, the IDE should not suggest a seemingly-innocuous clean-up that ends up having a potentially high cost.

**Comment by CryoMyst** [comment]
This seems to have been done for net222 branch.
https://github.com/JetBrains/resharper-unity/commit/6f40958d06c8f1208113589385e67f4b0ad6e79b

**Comment by citizenmatt** [comment]
Yes, this and #2319 are duplicates. Fixed in #2334, released as part of 2022.2
